doc/source/serve/deploy-many-models/multi-app.md

```diff
@@ -3,65 +3,68 @@
 
 In Ray 2.4+, deploying multiple independent Serve applications is supported. This user guide walks through how to generate a multi-application config file and deploy it using the Serve CLI, and monitor your applications using the CLI and the Ray Serve dashboard.
 
-## When to Use Multiple Applications
+## Context
+### Background 
 With the introduction of multi-application Serve, we walk you through the new concept of applications and when you should choose to deploy a single application versus multiple applications per cluster. 
 
 An application consists of one or more deployments. The deployments in an application are tied into a direct acyclic graph through [model composition](serve-model-composition). An application can be called via HTTP at the specified route prefix, and the ingress deployment handles all such inbound traffic. Due to the dependence between deployments in an application, one application is a unit of upgrade. 
 
-With that in mind, if you have many independent models each behind different endpoints, and you want to be able to easily add, delete, or update these models, then you should use multiple applications. Each model should then be deployed as a separate application. On the other hand, if you have ML logic and business logic distributed among separate deployments that all need to be executed for a single request, then you should use model composition to build a single application consisting of multiple deployments.
+### When to Use Multiple Applications
+Many use cases can be solved by either model composition or multi-application. However, both have their own individual benefits and can be used together.
+
+Suppose you have multiple models and/or business logic that all need to be executed for a single request. If they are living in one repository, then most likely they will be upgraded as a unit, so we recommend having all those deployments in one application.
+
+On the other hand, if there are logical groups among these models and/or business logic, e.g. where the groups of models still communicate with each other but live in different repositories, we recommend separating the models into applications. Since one application is a unit of upgrade, having multiple applications allows you to deploy many independent models (or groups of models) each behind different endpoints. You can then use easily add or delete applications from the cluster as well as upgrade applications independently of each other.
```

Comment:
Maybe add - Another common use-case for multiple applications is separate groups of models that may not communicate with each other but you want to co-host them for improving hardware utilization.